public/app/features/logs/components/logParser.ts

```diff
@@ -35,20 +35,30 @@ export const getAllFields = (
 export const createLogLineLinks = (hiddenFieldsWithLinks: FieldDef[]): FieldDef[] => {
   let fieldsWithLinksFromVariableMap: FieldDef[] = [];
   hiddenFieldsWithLinks.forEach((linkField) => {
-    linkField.links?.forEach((link: ExploreFieldLinkModel) => {
-      if (link.variables) {
-        const variableKeys = link.variables.map((variable) => {
-          const varName = variable.variableName;
-          const fieldPath = variable.fieldPath ? `.${variable.fieldPath}` : '';
-          return `${varName}${fieldPath}`;
-        });
-        const variableValues = link.variables.map((variable) => (variable.found ? variable.value : ''));
-        fieldsWithLinksFromVariableMap.push({
-          keys: variableKeys,
-          values: variableValues,
-          links: [link],
-          fieldIndex: linkField.fieldIndex,
-        });
+    linkField.links?.forEach((link: LinkModel | ExploreFieldLinkModel) => {
+      if ('variables' in link) {
+        if (link.variables.length > 0) {
+          // convert ExploreFieldLinkModel to LinkModel by omitting variables field
+          const fieldDefFromLink: LinkModel = {
+            href: link.href,
+            title: link.title,
+            origin: link.origin,
+            onClick: link.onClick,
+            target: link.target,
+          };
+          const variableKeys = link.variables.map((variable) => {
+            const varName = variable.variableName;
+            const fieldPath = variable.fieldPath ? `.${variable.fieldPath}` : '';
+            return `${varName}${fieldPath}`;
+          });
+          const variableValues = link.variables.map((variable) => (variable.found ? variable.value : ''));
+          fieldsWithLinksFromVariableMap.push({
+            keys: variableKeys,
+            values: variableValues,
+            links: [fieldDefFromLink],
+            fieldIndex: linkField.fieldIndex,
+          });
+        }
```

Comment:
```suggestion
      if ('variables' in link && link.variables.length > 0) {
        // convert ExploreFieldLinkModel to LinkModel by omitting variables field
        const fieldDefFromLink: LinkModel = {
          href: link.href,
          title: link.title,
          origin: link.origin,
          onClick: link.onClick,
          target: link.target,
        };
        const variableKeys = link.variables.map((variable) => {
          const varName = variable.variableName;
          const fieldPath = variable.fieldPath ? `.${variable.fieldPath}` : '';
          return `${varName}${fieldPath}`;
        });
        const variableValues = link.variables.map((variable) => (variable.found ? variable.value : ''));
        fieldsWithLinksFromVariableMap.push({
          keys: variableKeys,
          values: variableValues,
          links: [fieldDefFromLink],
          fieldIndex: linkField.fieldIndex,
        });
      }
```
Nit but improves readability.